### PR TITLE
fix in XYZ->Bins to reproduce binning of Miko

### PR DESCRIPTION
**Pull request by shahor02**

### CODE REVIEW

**Comment by shahor02** [comment]
@dr4kan : thanks for spotting the problem: the binning of Miko overshoots the pixel boundaries in X
and undershoots in Y (i.e. his highest bin boundary in Y stops short of the maxY of the pixel, so he 
extends the last Y bin beyond the uniform binning coverage. And the same happens for the lowest bin in Z.
Attachment shows in blue coordinate -> mapping of Miko (bin <0 on Z plot means that no response is retrieved) and in red - the AlpideSimResponse implementation (w/o bin range check, so also unphysical coordinates are assigned a binID).
This PR makes AlpideSimResponse to have exactly the same binning definition as in Miko's code.

image

**Comment by shahor02** [comment]
Hi, can this PR be merged? It is just a bug fix.

**Comment by sawenzel** [comment]
@dberzano : Is the macos build down?

File: Detectors/ITSMFT/common/simulation/include/ITSMFTSimulation/AlpideSimResponse.h [hunk 1]
```diff
@@ -104,9 +107,9 @@ class AlpideSimResponse
   float getYMax() const { return mYMax; }
   float getZMin() const { return mZMin; }
   float getZMax() const { return mZMax; }
-  float getStepX() const { return mStepInvX ? mStepInvX : 0.f; }
-  float getStepY() const { return mStepInvY ? mStepInvY : 0.f; }
-  float getStepZ() const { return mStepInvZ ? mStepInvZ : 0.f; }
+  float getStepX() const { return mStepInvX ? 1./mStepInvX : 0.f; }
```
Comment (sawenzel) [inline Detectors/ITSMFT/common/simulation/include/ITSMFTSimulation/AlpideSimResponse.h]: would it be reasonable storing `mStepX` directly as a member? (avoids division)

Comment (shahor02) [reply Detectors/ITSMFT/common/simulation/include/ITSMFTSimulation/AlpideSimResponse.h]: It is not used in any algorithm loop, just a method to know the segmentation (if needed).

Comment (sawenzel) [reply Detectors/ITSMFT/common/simulation/include/ITSMFTSimulation/AlpideSimResponse.h]: fair enough

**Pull request merged**